src/base/scanfoldersmodel.cpp

```diff
@@ -342,7 +342,7 @@ void ScanFoldersModel::configure()
 {
     QVariantHash dirs = Preferences::instance()->getScanDirs();
```

Comment:
Is dirs should be modifiable? Otherwise just declare it as const and then begin() returns const iterator.
The same comment is about some other lines affected by this PR.

Reply:
@glassez That's another way to go. I'm Ok with that.

Reply:
@luis-pereira, so please fix all these places.